openasip/doc/man/OpenASIP/OpenASIP.tex

```diff
@@ -4136,6 +4136,61 @@ \subsection{Defining a custom operation using an HDL snippet}
 but same information should still be printed to stdout meaning that the algorithm
 was executed correctly.
 
+\subsection{QEMU-OpenASIP}
+Simulating custom RISC-V instructions for larger programs may be too slow for 
+some use cases, and for that we have an extension to QEMU to link with libopenasip.
+The basic idea is to execute standard RISC-V instructions with QEMU, and send all of the
+custom instructions the program has for OpenASIP to resolve. You can build our QEMU fork with:
+
+\begin{verbatim}
+$ git clone https://github.com/cpc/qemu-openasip.git
+$ cd qemu-openasip 
+$ ./configure --target-list=riscv64-softmmu,riscv32-softmmu
+$ make -j${nproc}
+\end{verbatim}
+
+And then you can run any program with custom OSAL instructions with the QEMU's
+RISC-V 'virt' machine by specifying the new machine parameter \textbf{openasip\_machine}.
+For example:
+
+\begin{verbatim}
+$ ./build/qemu-system-riscv32 \
+    -machine virt,oasip_machine=/path/to/machine.adf \
+    -bios none \
+    -nographic \
+    -no-reboot \
+    -kernel /path/to/program.elf
+\end{verbatim}
+
+The program (or "kernel") you want to execute should be a .elf file if you are targeting 
+baremetal, but for example RISC-V linux with a BusyBox works too. The process 
```

Comment:
Does it work? Does it then simulate our .adf since the RISC-V cores we generate do not contain instructions/microarchitecture needed for usual OS services.